js/interface.js

```diff
@@ -919,7 +946,9 @@ function saveCurrentData() {
     var clientIdMap = _.zipObject(clientIds, ids);
 
     cacheOriginalEntries(entries, clientIdMap);
-    table.setData({ columns: columns, rows: entries });
+    HistoryStack.columnsInfo.reset();
+
+    setTableData({ columns, rows: entries });
```

Comment:
This says that every time changes are saved, we are reinitialising the entire Handsontable instance, instead of using `hot.loadData()` to update existing data. Is this not a bit heavy-handed and potentially creates unnecessary initialisation time?

Reply:
@tonytlwu Yes, indeed, and it's done on purpose since Handsontable otherwise cannot handle the update of columns list with trailing column correctly. As we have pagination now, it's way cheaper to reinitialize Handsontable.